cmd/prepare.go

```diff
@@ -63,14 +64,20 @@ func prepareTrack(id string) error {
 		return err
 	}
 
-	payload := &prepareTrackPayload{}
-	res, err := client.Do(req, payload)
+	res, err := client.Do(req)
 	if err != nil {
 		return err
 	}
+	defer res.Body.Close()
+
+	payload := &prepareTrackPayload{}
+
+	if err := json.NewDecoder(res.Body).Decode(payload); err != nil {
+		return fmt.Errorf("unable to parse API response - %s", err)
+	}
 
 	if res.StatusCode != http.StatusOK {
-		return errors.New("api call failed")
+		return errors.New(payload.Error.Message)
```

Comment:
Without knowing too much about the API response. If for whatever reason Error is empty, or not provided in the response we will return an error with no message. If this is a non issue then we can ignore this observation

Reply:
For now I'm ok with ignoring it. We're going to have to go through all the error handling and messaging with a fine-tooth comb, but I'd like to get the basic shape in place so that we have something reasonable to build off of.